controllers/keda/metrics_adapter_controller.go

```diff
@@ -18,20 +18,69 @@ package keda
 
 import (
 	"context"
+	"sync"
 
-	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
-	"github.com/kedacore/keda/v2/pkg/scaling"
+	"k8s.io/apimachinery/pkg/api/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/custom-metrics-apiserver/pkg/provider"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/scaling"
 )
 
 type MetricsScaledObjectReconciler struct {
-	ScaleHandler scaling.ScaleHandler
+	Client                  client.Client
+	ScaleHandler            scaling.ScaleHandler
+	ExternalMetricsInfo     *[]provider.ExternalMetricInfo
```

Comment:
same question here RE the pointer to slice. I think @coderanger mentioned this [below](https://github.com/kedacore/keda/pull/2279/files#r748162610) - just using a slice wouldn't be much less performant. it can also save a little bit of complexity

Reply:
Same as above